editor/src/components/editor/actions/actions.tsx

```diff
@@ -2008,9 +2012,36 @@ export const UPDATE_FNS = {
     }
   },
   UPDATE_GITHUB_CHECKSUMS: (action: UpdateGithubChecksums, editor: EditorModel): EditorModel => {
+    const githubChecksums = action.checksums != null ? { ...action.checksums } : null
+    const assetChecksums = { ...editor.assetChecksums }
+    if (githubChecksums != null) {
+      // patch checksums
+      Object.keys(editor.assetChecksums).forEach((k) => {
+        if (githubChecksums[k] == undefined) {
+          githubChecksums[k] = editor.assetChecksums[k] // local, non-committed checksums win
+        } else {
+          assetChecksums[k] = githubChecksums[k] // remote sha checksums win
+        }
+      })
+    }
+    return {
+      ...editor,
+      githubChecksums: githubChecksums,
+      assetChecksums: assetChecksums,
+    }
+  },
+  SET_ASSET_CHECKSUM: (action: SetAssetChecksum, editor: EditorModel): EditorModel => {
+    const assetChecksums: GithubChecksums =
+      editor.assetChecksums == null ? {} : { ...editor.assetChecksums }
+    if (action.checksum == null) {
+      delete assetChecksums[action.filename]
+    } else {
+      assetChecksums[action.filename.replace(/^\.\//, '/')] = action.checksum
+    }
```

Comment:
Looks like there's a bug here, as we're using a potentially different key when updating a value in the object vs when deleting from it.

Reply:
yep that's true, but I'm unable to find out who is not setting it correctly. maybe a pair of extra eyes could help here 👀

Reply:
actually found it → https://github.com/concrete-utopia/utopia/blob/65fa7b54943af7f97886464225e4054376c7a211/website-next/components/common/server.ts#L68-L70

Reply:
fixed in this one https://github.com/concrete-utopia/utopia/pull/2933/commits/655d01e91cc6c98192d7834a0ff078ee91d6a647